lib/src/view/analysis/analysis_board.dart

```diff
@@ -1,46 +1,143 @@
+import 'package:lichess_mobile/src/view/analysis/game_analysis_board.dart';
+import 'package:lichess_mobile/src/view/broadcast/broadcast_game_screen.dart';
+import 'package:lichess_mobile/src/view/study/study_screen.dart';
+import 'package:lichess_mobile/src/widgets/pgn.dart';
 
-class AnalysisBoard extends AbstractAnalysisBoard {
-  const AnalysisBoard({
-    required this.options,
-    required super.boardSize,
-    super.boardRadius,
-    this.shouldReplaceChildOnUserMove = false,
-  });
+/// An abstract widget that provides the common interface for three types of analysis boards:
+/// - [GameAnalysisBoard]
+/// - [BroadcastAnalysisBoard]
+/// - [StudyAnalysisBoard]
+abstract class AnalysisBoard extends ConsumerStatefulWidget {
+  const AnalysisBoard({super.key, required this.boardSize, this.boardRadius});
 
-  final AnalysisOptions options;
+  final double boardSize;
+  final BorderRadiusGeometry? boardRadius;
+}
 
-  final bool shouldReplaceChildOnUserMove;
+/// Abstract state class for analysis board widgets.
+abstract class AnalysisBoardState<
+  T extends AnalysisBoard,
+  AnalysisState extends CommonAnalysisState,
+  AnalysisPrefs extends CommonAnalysisPrefs
+>
+    extends ConsumerState<T> {
+  AnalysisState get analysisState;
 
-  @override
-  ConsumerState<AnalysisBoard> createState() => AnalysisBoardState();
-}
+  AnalysisPrefs get analysisPrefs;
 
-class AnalysisBoardState
-    extends AbstractAnalysisBoardState<AnalysisBoard, AnalysisState, AnalysisPrefs> {
-  @override
-  AnalysisState get analysisState =>
-      ref.watch(analysisControllerProvider(widget.options)).requireValue;
+  /// Whether the annotations should be shown on the board.
+  bool get showAnnotations;
 
-  @override
-  AnalysisPrefs get analysisPrefs => ref.watch(analysisPreferencesProvider);
+  void onUserMove(NormalMove move);
 
-  @override
-  bool get showAnnotations =>
-      analysisState.isComputerAnalysisAllowed &&
-      analysisState.isServerAnalysisEnabled &&
-      analysisPrefs.showAnnotations;
+  void onPromotionSelection(Role? role);
 
-  @override
-  void onUserMove(NormalMove move) => ref
-      .read(analysisControllerProvider(widget.options).notifier)
-      .onUserMove(move, shouldReplace: widget.shouldReplaceChildOnUserMove);
+  /// For the study board to set a different fen if the position is `null`.
+  String get fen => analysisState.currentPosition!.fen;
+
+  /// For the study board to clear shapes when switching to a new chapter.
+  void clearShapes() {}
+
+  /// For the study board to add pgn shapes and variations arrows.
+  ISet<Shape> get extraShapes => ISet();
+
+  /// Set of shapes drawn by the user on the board (arrows, circle).
+  ISet<Shape> userShapes = ISet();
 
   @override
-  void onPromotionSelection(Role? role) =>
-      ref.read(analysisControllerProvider(widget.options).notifier).onPromotionSelection(role);
+  Widget build(BuildContext context) {
+    clearShapes();
```

Comment:
I missed that in the last review, and I don't think we should do it like that. The name itself is misleading because this function calls `ref.listen`;

What about removing `clearShapes` from `AnalysisBoardState` and do:
```dart
class _StudyAnalysisBoardState extends State<...> {
   ...

   @override
   build() {
          // Clear shapes when switching to a new chapter.
    // This avoids "leftover" shapes from the previous chapter when the engine has not evaluated the new position yet.
    ref.listen(studyControllerProvider(widget.id).select((state) => state.hasValue), (prev, next) {
      if (prev != next) {
        setState(() {
          userShapes = ISet();
        });
      }
    });
    
     return super.build();
   }
}